From: Bastian Blank <waldi@debian.org>
Subject: Allow access to sensitive SysRq keys to be restricted by default
Date: Sun, 14 Feb 2010 16:11:35 +0100

Add a Kconfig variable to set the initial value of the Magic
SysRq mask (sysctl: kernel.sysrq).

Index: linux-3.2.46/include/linux/sysrq.h
===================================================================
--- linux-3.2.46.orig/include/linux/sysrq.h	2013-07-26 18:56:36.000000000 +0000
+++ linux-3.2.46/include/linux/sysrq.h	2013-07-26 19:17:11.000000000 +0000
@@ -18,7 +18,7 @@
 #include <linux/types.h>
 
 /* Enable/disable SYSRQ support by default (0==no, 1==yes). */
-#define SYSRQ_DEFAULT_ENABLE	1
+#define SYSRQ_DEFAULT_ENABLE	CONFIG_MAGIC_SYSRQ_DEFAULT_MASK
 
 /* Possible values of bitmask for enabling sysrq functions */
 /* 0x0001 is reserved for enable everything */
Index: linux-3.2.46/lib/Kconfig.debug
===================================================================
--- linux-3.2.46.orig/lib/Kconfig.debug	2013-07-26 18:56:36.000000000 +0000
+++ linux-3.2.46/lib/Kconfig.debug	2013-07-26 19:17:11.000000000 +0000
@@ -62,6 +62,14 @@
 	  keys are documented in <file:Documentation/sysrq.txt>. Don't say Y
 	  unless you really know what this hack does.
 
+config MAGIC_SYSRQ_DEFAULT_MASK
+	hex "Default mask for Magic SysRq keys on the console"
+	depends on MAGIC_SYSRQ
+	default 1
+	help
+	  Specifies the default mask for the allowed SysRq keys.  This can be
+	  used to disable several sensitive keys by default.
+
 config STRIP_ASM_SYMS
 	bool "Strip assembler-generated symbols during link"
 	default n
